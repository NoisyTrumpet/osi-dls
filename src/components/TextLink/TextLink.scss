.text-link {
  text-decoration: none;

  &.has-icon {
    align-items: center;
    justify-content: center;
    display: inline-flex;
    margin-bottom: 3px;
  }

  &.icon-left {
    flex-direction: row-reverse;

    .icon {
      margin-right: 8px;
    }
  }

  &.icon-right {
    .icon {
      margin-left: 8px;
    }
  }

  &:disabled {
    @include state-does-not-exist;
  }
}
