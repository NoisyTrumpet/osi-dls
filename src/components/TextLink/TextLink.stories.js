import React from "react";
import { storiesOf } from "@storybook/react";
import { select } from "@storybook/addon-knobs";
import { withSmartKnobs } from "storybook-addon-smart-knobs";
import { TextLink } from "./";
import iconArray from "../Icon/helpers/getIconArray";

const parameters = {
  component: TextLink,
  componentSubtitle: "Utilize <TextLink /> to present hyperlinks",
  jest: ["TextLink"]
};

storiesOf("components/TextLink", module)
  .addParameters(parameters)
  .addDecorator(withSmartKnobs({ ignoreProps: ["icon"] }))
  .add("Default", () => (
    <TextLink
      to="https://osilife.com/"
      title="Text Link"
      icon={select("icon", iconArray, "placeholder-bold")}
    />
  ));

storiesOf("components/TextLink", module)
  .addParameters(parameters)
  .addDecorator(withSmartKnobs({ ignoreProps: ["icon"] }))
  .add("Quick Link", () => (
    <TextLink
      to="https://osilife.com/"
      title="Text Link"
      defaultClassOverride="btn btn-tertiary"
      typoVariant="cta-tertiary"
      color="primary"
      icon={select("icon", iconArray, "placeholder-bold")}
    />
  ));

storiesOf("components/TextLink", module)
  .addParameters(parameters)
  .addDecorator(withSmartKnobs({ ignoreProps: ["icon"] }))
  .add("icon-right", () => (
    <TextLink
      to="https://osilife.com/"
      title="Text Link"
      icon="placeholder-bold"
      iconPos="right"
    />
  ));

storiesOf("components/TextLink", module)
  .addParameters(parameters)
  .addDecorator(withSmartKnobs({ ignoreProps: ["icon"] }))
  .add("icon-left", () => (
    <TextLink
      to="https://osilife.com/"
      title="Text Link"
      icon="placeholder-bold"
      iconPos="left"
    />
  ));
