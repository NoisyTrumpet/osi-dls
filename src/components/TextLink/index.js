export { default as TextLink } from './TextLink'; // eslint-disable-line import/prefer-default-export
