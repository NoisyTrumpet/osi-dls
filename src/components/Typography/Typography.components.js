import classNames from 'classnames';

/* TypographyDetails returns 2 props: 'html_tag' and 'className' */
const TypographyDetails = (defaultTag, className) => ({ defaultTag, className: classNames('typography', className) });

/*
TypographyComponents is a Key / Value pair
- Key: a valid type of <Typography />
- Value: a TypographyDetails object (defined above)
*/
const TypographyComponents = {
  'headline-1':
    TypographyDetails('h1', 'typography-headline typography-headline-1'),
  'headline-2':
    TypographyDetails('h2', 'typography-headline typography-headline-2'),
  'headline-3':
    TypographyDetails('h3', 'typography-headline typography-headline-3'),
  'headline-4':
    TypographyDetails('h4', 'typography-headline typography-headline-4'),
  'headline-5':
    TypographyDetails('h5', 'typography-headline typography-headline-5'),
  'headline-6':
    TypographyDetails('h6', 'typography-headline typography-headline-6'),
  'price-unit':
    TypographyDetails('span', 'typography-price-unit'),
  'price-unit-strikethrough':
    TypographyDetails('span', 'typography-price-unit-strikethrough'),
  'price-sale':
    TypographyDetails('span', 'typography-price-sale'),
  cta:
    TypographyDetails('span', 'typography-cta'),
  'cta-tertiary':
    TypographyDetails('span', 'typography-cta-tertiary'),
  subtitle:
    TypographyDetails('span', 'typography-subtitle'),
  'field-label':
    TypographyDetails('span', 'typography-field-label'),
  body:
    TypographyDetails('span', 'typography-body'),
  'body-medium':
    TypographyDetails('span', 'typography-body typography-body-medium'),
  eyebrow:
    TypographyDetails('span', 'typography-eyebrow'),
  'text-link':
    TypographyDetails('span', 'typography-text-link'),
  'input-text':
    TypographyDetails('span', 'typography-input-text'),
  caption:
    TypographyDetails('span', 'typography-caption'),
  helper:
    TypographyDetails('span', 'typography-helper'),
  breadcrumb:
    TypographyDetails('span', 'typography-breadcrumb'),
  srt:
    TypographyDetails('span', 'screen-reader-text'),
  licensing:
    TypographyDetails('span', 'typography-licensing'),
  subtext:
    TypographyDetails('span', 'typography-subtext'),
};

export default TypographyComponents;
