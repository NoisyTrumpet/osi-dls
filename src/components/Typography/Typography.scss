// Body
.typography-body {
  color: map-get($colors, 'gray5');
  font-size: rem(15px);
  letter-spacing: 0px;
  @include fontSans($gtAmerican, $gtAmerican, 'sans-serif', 84, 700);
}

.typography-breadcrumb,
.typography-body-medium,
.typography-cta-tertiary,
.typography-caption,
.typography-eyebrow,
.typography-helper,
.typography-input-text,
.typography-licensing,
.typography-subtext,
.typography-subtitle,
.typography-field-label,
.typography-text-link,
.typography-price-unit,
.typography-price-unit-strikethrough,
.typography-price-sale {
  @include fontSans($gtAmerican, $gtAmerican, 'sans-serif', 114, 700);
}

// Call To Action
.typography-cta {
  color: map-get($colors, 'primary');
  font-size: rem(14px);
  letter-spacing: rem(0.7px);
  text-transform: uppercase;
  @include fontSans($gtAmerican, $gtAmerican, 'sans-serif', 700, 700);
}

// Call To Action - Tertiary
.typography-cta-tertiary {
  color: map-get($colors, 'primary');
  font-size: rem(14px);
  letter-spacing: 0px;
}

//Caption
.typography-caption {
  color: map-get($colors, 'util-error');
  font-size: rem(13px);
  letter-spacing: 0px;
}

// Eyebrow
.typography-eyebrow {
  color: map-get($colors, 'gray5');
  font-size: rem(14px);
  text-transform: uppercase;
  letter-spacing: rem(0.8px);
}

// Headline
.typography-headline {
  color: map-get($colors, 'primary');
  @include fontSerif;
}

.typography-headline-1 {
  font-size: rem(64px);
  letter-spacing: rem(-1px);
}

.typography-headline-2 {
  font-size: rem(46px);
  letter-spacing: rem(-0.8px);
}

.typography-headline-3 {
  font-size: rem(36px);
  letter-spacing: rem(-0.5px);
}

.typography-headline-4 {
  font-size: rem(31px);
  letter-spacing: rem(-0.4px);
}

.typography-headline-5 {
  font-size: rem(24px);
  letter-spacing: rem(-0.2px);
}

.typography-headline-6 {
  font-size: rem(22px);
  letter-spacing: 0px;
}

// Price
.typography-price-unit {
  font-size: rem(22px);
}

.typography-price-unit-strikethrough {
  text-decoration: line-through;
  font-size: rem(16px);
}

.typography-price-sale {
  font-size: rem(22px);
}

// Helper
.typography-helper {
  color: map-get($colors, 'gray5');
  font-size: rem(14px);
  letter-spacing: 0px;
}

// Input Text
.typography-input-text {
  color: map-get($colors, 'gray5');
  font-size: rem(16px);
  letter-spacing: 0px;
}

// Subtitle
.typography-subtitle {
  color: map-get($colors, 'primary');
  font-size: rem(18px);
  letter-spacing: 0px;
}

// Field Label
.typography-field-label {
  color: map-get($colors, 'gray4');
  font-size: rem(15px);
  letter-spacing: rem(0.3px);
}

// Text Link
.typography-text-link {
  color: map-get($colors, 'accentB');
  font-size: rem(15px);
  letter-spacing: 0px;
  text-decoration: underline;
}

// Breadcrumb
.typography-breadcrumb,
.typography-licensing,
.typography-subtext {
  color: map-get($colors, 'gray5');
  font-size: rem(13px);
  letter-spacing: 0px;
}

// Screen Reader Text
.screen-reader-text {
  @extend %screen-reader-text;
}

// Colors
@each $name, $color in $colors {
  .color-#{$name},
  .color-#{$name}:before {
    color: $color;
  }
}
