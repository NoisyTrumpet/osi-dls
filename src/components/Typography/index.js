export { default as Typography } from './Typography'; // eslint-disable-line import/prefer-default-export
