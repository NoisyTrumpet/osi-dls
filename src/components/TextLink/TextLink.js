import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Typography } from '../Typography';
import { Icon } from '../Icon';
import './TextLink.scss';
// TODO: Link types (internal / ext-hallmark / external)

// Utilize `TextLink` to navigate around the application, or to link to an external site
const TextLink = ({
  defaultClassOverride,
  addClass,
  to,
  title,
  color,
  icon,
  iconPos,
  typoVariant,
  target,
  rel,
}) => {
  const defaultClass = (defaultClassOverride || 'text-link');
  const iconClass = iconPos ? (`has-icon icon-${iconPos}`) : '';

  const classes = classNames(
    defaultClass,
    addClass,
    iconClass,
  );

  return (
    <a href={to} className={classes} target={target} rel={rel}>
      <Typography variant={typoVariant} color={color}>{title}</Typography>
      {
        iconPos
        && (
          <Icon
            name={icon}
            color={color || 'accentB'}
            size={14}
          />
        )
      }
    </a>
  );
};

TextLink.defaultProps = {
  icon: 'placeholder-bold',
  typoVariant: 'text-link',
};

TextLink.propTypes = {
  defaultClassOverride: PropTypes.string,
  addClass: PropTypes.string,
  to: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  color: PropTypes.string,
  iconPos: PropTypes.oneOf([
    'left',
    'right',
  ]),
  icon: PropTypes.string,
  typoVariant: PropTypes.oneOf([
    'text-link',
    'helper',
    'cta-tertiary',
    'cta',
    'body',
  ]),
  target: PropTypes.string,
  rel: PropTypes.string,
};

export default TextLink;
