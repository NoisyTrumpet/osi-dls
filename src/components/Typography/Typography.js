import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import TypographyComponents from './Typography.components';
import './Typography.scss';

/**
 * Utilize `Typography` to present text
 */
const Typography = ({ addClass, variant, htmlTagOverride, color, children, ...rest }) => {
  if (!(variant in TypographyComponents)) {
    return (null);
  }

  const {
    defaultTag,
    className,
  } = TypographyComponents[variant];

  const colorClass = {
    [`color-${color}`]: color,
  };

  const newProps = {
    className: classNames(className, colorClass, addClass),
    ...rest,
  };

  const tag = htmlTagOverride || defaultTag;
  return React.createElement(tag, newProps, children);
};

Typography.variants = Object.keys(TypographyComponents);

Typography.propTypes = {
  /** Variant determines the styling for the component */
  variant: PropTypes.oneOf([
    'headline-1',
    'headline-2',
    'headline-3',
    'headline-4',
    'headline-5',
    'headline-6',
    'price-unit',
    'price-unit-strikethrough',
    'price-sale',
    'cta',
    'cta-tertiary',
    'subtitle',
    'field-label',
    'body',
    'body-medium',
    'eyebrow',
    'text-link',
    'input-text',
    'caption',
    'helper',
    'breadcrumb',
    'srt',
    'licensing',
    'subtext',
  ]).isRequired,
  /** The content of the component */
  children: PropTypes.node,
  /** Overides the default tag that the component is rendered as */
  htmlTagOverride: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'p', 'span', 'address', 'div']),
  /** Color provides the component with a class name that affects the color of text */
  color: PropTypes.string,
  /** addClass appends a string to the component's className prop */
  addClass: PropTypes.string,
};

export default Typography;
