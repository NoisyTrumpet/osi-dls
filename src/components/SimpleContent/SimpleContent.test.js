import React from 'react';
import { render, screen } from '@testing-library/react';
import { toBeInTheDocument } from '@testing-library/jest-dom/matchers';
import '@testing-library/jest-dom/extend-expect';
import { SimpleContent } from './';
import { Typography } from '../Typography';
import { TextLink } from '../TextLink';

expect.extend({
  toBeInTheDocument,
});

describe('<SimpleContent />', () => {
  it('Should render', () => {
    const contentText = 'Test content';
    const content = [
      <p key={1}>
        <TextLink
          typo-variant="text-link"
          title="Test Link"
          to="https://www.hallmark.com"
        />
      </p>,
      <Typography variant="body" key={2}>
        {contentText}
      </Typography>,
    ];

    const title = 'Test Title';
    const icon = 'home';
    const subhead = 'Test Subhead';
    const { container } = render(
      <SimpleContent
        title={title}
        icon={icon}
        subhead={subhead}
        content={content}
      />,
    );
    expect(screen.getByText(title)).toBeInTheDocument();
    expect(screen.getByText(subhead)).toBeInTheDocument();
    expect(screen.getByText(contentText)).toBeInTheDocument();
    expect(screen.getByText(contentText)).toHaveClass('typography-body');
    expect(screen.getByTestId('icon')).toBeInTheDocument();
    expect(screen.getByTestId('icon')).toHaveClass(`icon-${icon}`);
    expect(container.querySelector('a')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('Should not render icon or subhead', () => {
    const title = 'Test Title';
    const subhead = 'Test Subhead';
    const content = [<button type="submit">button</button>];
    const { container } = render(<SimpleContent content={content} />);
    expect(screen.queryByText(title)).not.toBeInTheDocument();
    expect(screen.queryByText(subhead)).not.toBeInTheDocument();
    expect(screen.queryByTestId('icon')).not.toBeInTheDocument();
    expect(screen.getByRole('button')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
