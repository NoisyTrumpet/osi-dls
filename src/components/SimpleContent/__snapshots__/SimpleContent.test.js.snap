// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleContent /> Should not render icon or subhead 1`] = `
<div>
  <div
    class="simplecontent-wrapper"
  >
    <div
      class="simplecontent-body"
    >
      <div
        class="simplecontent-content"
      >
        <button
          type="submit"
        >
          button
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SimpleContent /> Should render 1`] = `
<div>
  <div
    class="simplecontent-wrapper"
  >
    <h3
      class="typography typography-body typography-body-medium"
    >
      Test Title
    </h3>
    <div
      class="simplecontent-body"
    >
      <div
        class="simplecontent-icon"
      >
        <span
          aria-hidden="true"
          class="icon icon-home icon-16"
          data-testid="icon"
        />
      </div>
      <div
        class="simplecontent-content"
      >
        <span
          class="typography typography-body"
        >
          Test Subhead
        </span>
        <p>
          <a
            class="text-link"
            href="https://www.hallmark.com"
          >
            <span
              class="typography typography-text-link"
            >
              Test Link
            </span>
          </a>
        </p>
        <span
          class="typography typography-body"
        >
          Test content
        </span>
      </div>
    </div>
  </div>
</div>
`;
